docs/user-guide/debugging-services.md

```diff
@@ -555,6 +555,25 @@ There are three things to check:
   Engine doesn't do port remapping, so if your application serves on 8080,
   the `containerPort` field needs to be 8080.
 
+### A Pod cannot reach itself via Service IP
+
+This mostly happens in `iptables` mode. There is `--hairpin-mode` flag in `Kubelet`.
+In order to allow endpoints of a Service to loadbalance back to themselves
+if they should try to access their own Service, `--hairpin-mode` flag has to be set
+to `haripin-veth` or `promiscuous-bridge`.
+
+The common steps to trouble shoot this are as follows:
+
+* Ensure `--hairpin-mode` is set to `haripin-veth` or `promiscuous-bridge`.
+
+* Check if there is any log lines with key word `hairpin` in kubelet.log. The log is usually
+located at `/var/log/kubelet.log` on node. There should be log lines indicating the effective
+hairpin mode. Please be noted that the effective hairpin mode may not match `--hairpin-mode` flag.
+
+* If the effective hairpin mode is `hairpin-veth`, ensure `Kubelet` has the permission to operate in `/sys` on node.
```

Comment:
tell: `for intf in /sys/devices/virtual/net/cbr0/brif/*; do cat $intf/hairpin_mode; done`